airflow/operators/docker_operator.py

```diff
@@ -93,86 +72,136 @@ class DockerOperator(BaseOperator):
+        https://goo.gl/65NcDi
+
+        .. code-block:: python
+
+            host_config = { 'binds': ['/home/some_path:/container/some_path'],
+                            'network_mode': 'bridge',
+                            'shm_size': 1000,
+                            'cpu_shares': 2048,
+                            'mem_limit': None,
+                            'auto_remove': False,
+                            'dns': None,
+                            'dns_search': None
+            }
     """
-    template_fields = ('command', 'environment',)
+    template_fields = ('container_config', 'host_config',)
     template_ext = ('.sh', '.bash',)
 
     @apply_defaults
     def __init__(
             self,
-            image,
             api_version=None,
-            command=None,
             cpus=1.0,
             docker_url='unix://var/run/docker.sock',
-            environment=None,
             force_pull=False,
-            mem_limit=None,
-            network_mode=None,
             tls_ca_cert=None,
             tls_client_cert=None,
             tls_client_key=None,
             tls_hostname=None,
             tls_ssl_version=None,
             tmp_dir='/tmp/airflow',
-            user=None,
-            volumes=None,
-            working_dir=None,
+            xcom_push=False,
             xcom_all=False,
             docker_conn_id=None,
-            dns=None,
-            dns_search=None,
-            auto_remove=False,
-            shm_size=None,
+            container_config=None,
+            host_config=None,
+
             *args,
             **kwargs):
 
         super(DockerOperator, self).__init__(*args, **kwargs)
         self.api_version = api_version
-        self.auto_remove = auto_remove
-        self.command = command
         self.cpus = cpus
-        self.dns = dns
-        self.dns_search = dns_search
         self.docker_url = docker_url
-        self.environment = environment or {}
         self.force_pull = force_pull
-        self.image = image
-        self.mem_limit = mem_limit
-        self.network_mode = network_mode
         self.tls_ca_cert = tls_ca_cert
         self.tls_client_cert = tls_client_cert
         self.tls_client_key = tls_client_key
         self.tls_hostname = tls_hostname
         self.tls_ssl_version = tls_ssl_version
         self.tmp_dir = tmp_dir
-        self.user = user
-        self.volumes = volumes or []
-        self.working_dir = working_dir
+        self.xcom_push_flag = xcom_push
         self.xcom_all = xcom_all
         self.docker_conn_id = docker_conn_id
-        self.shm_size = shm_size
-        if kwargs.get('xcom_push') is not None:
-            raise AirflowException("'xcom_push' was deprecated, use 'BaseOperator.do_xcom_push' instead")
+        self.container_config = container_config
+        self.host_config = host_config or {}
 
         self.cli = None
         self.container = None
 
+    def __set_default_params_in_host_config(self):
```

Comment:
I would recommend renaming both `__set_default_params_in_host_config` and `__set_default_params_in_container_config` to `__get_default_params_in_host_config` and `__get_default_params_in_container_config`. Since they seem to act more as a getter than a getter.

Reply:
thanks Kevin for your feedback, yes, I have added the following 2 methods in the class: 
```suggestion
    def __set_default_params_in_host_config(self):
    ....
```
```suggestion
   def __set_default_params_in_container_config(self):
   ....
```
I agree with you, I use these 2 private method to get the value of the arguments in my class, but what I implicitly  trying to do is to verify if the default required argument are well provided when calling a class, if not, these two methods set the required arguments with the default values. probably I need to change the name of this 2 methods to something more clear enough to not confuse them with `set` or `get` method. what do you think @kevincharbonneau, @galak75

Reply:
et pourquoi pas `__sanitize_host_config(self)` et `__sanitize_container_config(self)` ?